jsonschema/exceptions.py

```diff
@@ -61,14 +70,18 @@ def __init__(
         for error in context:
             error.parent = self
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f"<{self.__class__.__name__}: {self.message!r}>"
 
-    def __str__(self):
+    def __str__(self) -> str:
         essential_for_verbose = (
             self.validator, self.validator_value, self.instance, self.schema,
         )
-        if any(m is _unset for m in essential_for_verbose):
+        if (
+            any(m is _unset for m in essential_for_verbose)
+            or self._word_for_schema_in_error_message is None
```

Comment:
(Same here, I'd rather not add this, `_Error` is not actually meant to be instantiated, it's just an interface, though a messy one clearly with concrete behavior in it. Some day maybe it should instead be a function.)

Reply:
See my comment above!